### ISSUE TITLE
poly1d hangs (migrated from Trac #396)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#1003]
Original ticket http://projects.scipy.org/numpy/ticket/396
Reported 2006-12-05 by trac user joris, assigned to unknown.

from numpy import *

p = poly1d([nan,nan,1], r=0)

p.r

The last command makes python hang, not even CTRL-C helps.
In practice you don't give of course deliberately [nan,nan,1]
as coefficients, but earlier in the code something may have
gone wrong, so that the coefficients are not what is meant. 
I would expect then an ArithmeticError of poly1d rather than hanging.

Versions: Python 2.5 + Numpy 1.0

**Comment by thouis** [comment]
Comment in Trac by atmention:alberts, 2006-12-05

I can't reproduce this hang with NumPy 1.0.2.dev3476 running with Python 2.4.3 on Windows XP. Can you build NumPy from SVN and see if you still get this problem? Or try NumPy 1.0.1, it's pretty recent.

```
#!python
In [11]: from numpy import *

In [12]: p = poly1d([nan,nan,1], r=0)

In [13]: p
Out[13]: Warning: invalid value encountered in absolute
Warning: invalid value encountered in reduce
Warning: invalid value encountered in reduce
poly1d([-1.#IND, -1.#IND,  1.    ])
```

**Comment by thouis** [comment]
Comment in Trac by atmention:alberts, 2006-12-06

Ah, {{{p.r}}} is needed to make it hang. Reproduced on the configuration mentioned above.

**Comment by thouis** [comment]
Comment in Trac by atmention:teoliphant, 2006-12-06

The underlying LAPACK calls can't handle non-finite floating-point numbers (there is probably a loop somewhere that ends up comparing nan to nan and never finishes).  

In SciPy we placed a check for finite values but this is not in NumPy.  So, you will see the error whenever you don't have SciPy installed (because then the NumPy eig function will be called which hangs). 

I've added to SVN a check for finite values for the functions eig and eigvals (the others seemed to handle infs and nans O.K. --- but they should be checked).
